### ISSUE TITLE
Incorrect formatting of closure parameters

### ISSUE THREAD

**Issue by lawliet89** [rust-lang-nursery/rustfmt#1694]
In rustfmt 0.9.0, it produces incorrect Rust code while attempting to format some closure arguments as seen in this [diff](https://github.com/lawliet89/pr_demon/commit/31a29b8bf284ec161ddb84b607cda11f06f6e164#diff-bee84a580bea20166ade6d6a1a0a5f31R78).

**Comment by nrc** [comment]
```
-        Box::new(move |_reference : _, _target_reference : _, _oid : _, _target_oid : _| {
-                     format!("refs/pull/{}/merge", pr_id)
-                 })
+        Box::new(move |_reference, : _
+              _target_reference, : _
+              _oid, : _
+              _target_oid : _| {
+            format!("refs/pull/{}/merge", pr_id)
+        })
```

Commas are before the type, not after

**Comment by nrc** [comment]
It also looks like we ought to be able to fit the whole `| ... |` on one line

**Comment by topecongiro** [comment]
Closed via #1701.
